### PR TITLE
Refactorings in disasm.c

### PR DESCRIPTION
**Pull request by khrs**
According to comment in disasm.c
"TODO: put RCore inside RDisasmState and rename all functions to be ds_XXX"

### CODE REVIEW

File: libr/core/disasm.c [hunk 1]
```diff
@@ -869,14 +870,15 @@ static int var_comparator(const RAnalVar *a, const RAnalVar *b){
...
+static void ds_show_functions(RDisasmState *ds) {
 	RAnalFunction *f;
+	RCore *core = ds->core;
 	bool demangle, call;
 	const char *lang;
 	char *fcn_name;
 	char *sign;
 
-	if (!core || !ds || !ds->show_functions) {
+	if (!ds->show_functions) {
```
Comment (ret2libc) [inline libr/core/disasm.c]: why did you remove these checks?

Comment (khrs) [inline libr/core/disasm.c]: Plz corect me if i am wrong but this checks should belong to public api (aka r_core_* functions) not to internal ones.

BTW: They was only on this function nothing else. It is smth special about this function i am missing?

Comment (ret2libc) [inline libr/core/disasm.c]: No it's ok! :) The important thing is that it's removed for a right reason! Thanks

**Comment by radare** [comment]
Can you squash and rebase the pr? Domt merge in a branch you are pullrequesting

**Comment by khrs** [comment]
I will send this PR again. This time properly from seperated branch ;)
